### ISSUE TITLE
Faux Duplicates After Group Conversion

### ISSUE THREAD

**Issue by OkyDooky** [deckerst/aves#651]
**Describe the bug**
After converting multiple images, the output result shows many of them duplicated. The viewer will show the duplicates with the exact same file name, resolution, tags, etc.
If I go into any other gallery app or file browser, there will be no duplicates shown. I have no idea what's going on underneath, but restarting the app, phone, or both will not seem to fix the issue, even if I clear the cache after force-stopping the app.

**To Reproduce**
1. Select multiple files to convert
2. Convert and store in a different album
3. Open album
4. See these (non-existent) duplicates here and there.

**Screenshots**
![Screenshot_20230529-180718_Aves Libre](https://github.com/deckerst/aves/assets/59938024/493fa51d-54e1-4ea2-bb31-2b8c957b8756)

**System information and logs:**
Package: deckers.thibault.aves.libre
Installer: dev.imranr.obtainium
Aves version: 1.8.8-libre, build 9902
Flutter: stable 3.10.2
Android version: 13, API 33
Android build: lineage_ocean-userdebug 13 TQ1A.230205.002 eng.root.20230311.065633 dev-keys
Device: motorola moto g(7) power
Geocoder: ready
Mobile services: not available
System locales: en_US
Storage volumes: /storage/emulated/0/, /storage/3062-3833/
Storage grants: /storage/emulated/0/Movies/, /storage/3062-3833/Movies/, /storage/emulated/0/Pictures/, /storage/emulated/0/Documents/, /storage/emulated/0/DCIM/, /storage/3062-3833/Pictures/, /storage/emulated/0/Download/Pixel 4 Downloads and Photos/, /storage/emulated/0/Download/Image Converter/
Error reporting: false
[aves-logs-20230529_180800.txt](https://github.com/deckerst/aves/files/11594853/aves-logs-20230529_180800.txt)

**Additional context**
I've had this happen a few times in previous versions, as well. Deleting the duplicates seems to be guess work as to which is the real original and which is its ghost clone. It did seem to clear up after some time, from one conversion, but I'm unsure what finally did it. :/

**Comment by OkyDooky** [comment]
The duplicates also show up in the album's total item count.
For reference, there was one image there before this last operation and I converted 98 images.
![Screenshot_20230529-182023_Aves Libre](https://github.com/deckerst/aves/assets/59938024/533bebfb-210e-4f47-8941-f09fc8091dca)

**Comment by deckerst** [comment]
Thanks for the report. I'll take a look when i can. In the meantime, in order to get rid of the duplicates, what you can do is select all the pictures and move them somewhere. It should work for half of them and the duplicates would disappear.

**Comment by OkyDooky** [comment]
Thanks! I'll try that later and report back what happens.

**Comment by deckerst** [comment]
Can you reproduce the issue every time? I tried on my device and I can't reproduce it.

I don't know whether it's important, but could you let me know more details about the conversion?
- did you enable `write metadata`?
- what was the output format?

**Comment by OkyDooky** [comment]
I don't know if I can do it every time. I'll test it some more.
Yes to `write metadata` enabled, since I figured that was required to keep that tags.
JPG 96 quality was the target format and settings.

Also, when trying to move the files, to a different folder, this comes up:
![Screenshot_20230530-113953_Aves Libre](https://github.com/deckerst/aves/assets/59938024/8436c5c6-ba9f-422c-8674-921f64b4a0d2)

**Comment by OkyDooky** [comment]
Well, I tried moving them to a general folder that also only had one existing image in it and... it said "failed to move 78 images" and this is how many made it to the target folder:
![Screenshot_20230530-123426_Aves Libre](https://github.com/deckerst/aves/assets/59938024/f673aba2-1b8c-47e3-83a9-9bfea3bee092)
Apparently, only 20 survived. 😕 
I kept the originals, thankfully. So, I can try the conversion again, in the future.

**Comment by deckerst** [comment]
I did more tests:
- converting 30 items yielded 0 dups
- converting 40 items yielded ~13 dups
- converting 50 items yielded ~25 dups

The conversion parameters (quality, format, size, metadata writing) does not seem to matter. It seems to be a combination of number of items and bad timing.

As for getting rid of existing duplicates, moving them is __*supposed*__ to fail for duplicates, because it succeeds when it encounters the first one, and fails on the second. Are you sure you didn't have 78 duplicates? Anyway, for now I recommend to not convert too many items at once!

**Comment by OkyDooky** [comment]
That's interesting. 🤔 

>moving them is supposed to fail for duplicates

Yeah, the original number of files I converted was 98, which then bloated into 176 (plus the one originally in the Converted Images folder, as seen above). The moving process nullified the duplicates *and* failed to move 78 of the originals, since only 20 files made it to the final folder.
I haven't had an operation failure like that before, so I'm assuming it has to do with the duplicates issue (actually, I might have had a similar experience when I first encountered duplicating, several months or so ago).

But, yeah, I'll definitelt stick with smaller batches or use the converter app I used before for larger ones until this is figured out.

Thanks for looking into it!
